src/libstd/sys/unix/os.rs

```diff
@@ -59,6 +59,15 @@ pub fn errno() -> i32 {
         extern { fn __errno_location() -> *const c_int; }
         __errno_location()
     }
+    #[cfg(target_env = "newlib")]
+    fn errno_location() -> *const c_int {
+        extern {
+            fn __errno() -> *const c_int;
+        }
+        unsafe {
+            __errno()
+        }
```

Comment:
Could this use the same style as the instances above? Also if you're feeling ambitious feel free to change these all to `#[cfg_attr(..., link_name = "...")]` as it's probably the best way to write all this now.